storage/engine/mvcc.go

```diff
@@ -497,7 +497,12 @@ func MVCCGet(engine Engine, key roachpb.Key, timestamp roachpb.Timestamp, consis
 		return nil, nil, err
 	}
 
-	return mvccGetInternal(iter, key, metaKey, timestamp, consistent, txn, buf)
+	value, intents, err := mvccGetInternal(iter, key, metaKey, timestamp, consistent, txn, buf)
+	if value == &buf.value {
```

Comment:
No, I specifically do not want a deep copy. I'm going to add a call to `buf.value.Reset`, though to ensure the cached value has its fields cleared.

The returned `value` will either be `nil`, `&buf.value` or `buf.meta.Value`, the latter case occurs for data inlined inside the `MVCCMetadata`. There is no need to copy the pointer in that case.